src/js/index.js

```diff
@@ -46,28 +47,38 @@ function handleDifficulty(difficultyChanged) {
 	return false;
 }
 
-function handleTags() {
-	const tags = selectr.getValue();
+function updateUrl() {
+	let nextPath = window.location.pathname;
 
-	if (tags.length === 0) {
-		history.pushState(null, '', window.location.pathname);
-		return;
+	const tempParams = new URLSearchParams();
+	for (const [key, value] of search) {
+		if (value.trim()) {
```

Comment:
What's the point of using multiple instance of URLSearchParams?

If you use `if(!...) { search.delete(key) }` you would still get the same functionality without instantiating any new instances

Reply:
I tried but that didn't work well. My theory is that deleting a param while iterating over `search` caused the loop to skip over the next param.

Reply:
Makes sense...

```js
const badParams = [];
for (...) {
	if (!value.trim()) {
		badParams.push(key);
	}
}

badParams.forEach(param => search.delete(value));
```

Should reduce complexity